tests/test_streamingclient.py

```diff
@@ -54,33 +54,33 @@ def test_constructor_no_token_no_config(self):
             RevAiStreamingClient(None, example_config)
 
     def test_start_success(self, mock_streaming_client, mock_generator, capsys):
+        custom_vocabulary_id = 'mycustomvocabid'
         metadata = "my metadata"
-        url = mock_streaming_client.base_url + \
-            '?access_token={}'.format(mock_streaming_client.access_token) + \
-            '&content_type={}'. \
-            format(mock_streaming_client.config.get_content_type_string()) + \
-            '&user_agent={}'.format(quote('RevAi-PythonSDK/{}'.format(__version__), safe='')) + \
-            '&metadata={}'.format(quote(metadata))
+        query_dict = {
+            'access_token': mock_streaming_client.access_token,
+            'content_type': mock_streaming_client.config.get_content_type_string(),
+            'user_agent': 'RevAi-PythonSDK/{}'.format(__version__),
+            'custom_vocabulary_id': custom_vocabulary_id,
+            'metadata': metadata
+        }
         example_data = '{"type":"partial","transcript":"Test"}'
         example_connected = '{"type":"connected","id":"testid"}'
         if six.PY3:
             example_data = example_data.encode('utf-8')
             example_connected = example_connected.encode('utf-8')
-        data = [
-            [0x1, example_connected],
-            [0x1, example_data],
-            [0x8, b'\x03\xe8End of input. Closing']
-        ]
-        exp_responses = [
-            'Connected, Job ID : testid\n',
-            '{"type":"partial","transcript":"Test"}',
-            'Connection Closed. Code : 1000; Reason : End of input. Closing\n'
-        ]
+        data = [[0x1, example_connected],
+                [0x1, example_data],
+                [0x8, b'\x03\xe8End of input. Closing']]
+        exp_responses = ['Connected, Job ID : testid\n',
+                         '{"type":"partial","transcript":"Test"}',
+                         'Connection Closed. Code : 1000; Reason : End of input. Closing\n']
         mock_streaming_client.client.recv_data.side_effect = data
 
-        response_gen = mock_streaming_client.start(mock_generator(), metadata)
+        response_gen = mock_streaming_client.start(mock_generator(), custom_vocabulary_id, metadata)
 
-        mock_streaming_client.client.connect.assert_called_once_with(url)
+        mock_streaming_client.client.connect.assert_called_once()
```

Comment:
You should not use `assert_called_once()`. It was only introduced into unittest in python 3.6. It seems that previous versions of python don't catch that this is a problem because mocks in those versions (3.4 and 2.7) act as of you are just asking for a mock of some function called `assert_called_once`. You could probably confirm this by running 
```
what = mock_streaming_client.client.connect.assert_called_once()`
print(what)
```

See new in python 3.6: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once
Source: http://engineroom.trackmaven.com/blog/mocking-mistakes/

Reply:
Ahh, makes sense. I'll resort to measuring the length of the calls list or some equivalent then.